uniflow-core/src/test/kotlin/io/uniflow/core/rule/MockkTestRules.kt

```diff
@@ -0,0 +1,24 @@
+package io.uniflow.core.rule
+
+import io.mockk.clearAllMocks
+import io.mockk.unmockkAll
+import org.junit.ClassRule
+import org.junit.Rule
+import org.junit.rules.TestWatcher
+import org.junit.runner.Description
+
+/**
+ * Unmock all MockK mocks so that mocks don't leak between unit test classes.
+ * Apply this as a [@ClassRule][ClassRule].
+ */
+class MockkTestClassRule : TestWatcher() {
+    override fun finished(description: Description?) = unmockkAll()
+}
+
+/**
+ * Clear all MockK mocks so that mock state doesn't leak between unit test functions.
+ * Apply this as a [@Rule][Rule].
+ */
+class MockkTestRule : TestWatcher() {
+    override fun finished(description: Description?) = clearAllMocks()
+}
```

Comment:
Hello @erikhuizinga ,

what bring this rule?

Reply:
MockK can mock stuff permanently until you actively clear (reset mock state, e.g. function call count) or unmock (truly reset from MockK memory) it. The danger is that mocked behaviour leaks from one test to another! That is why `clearAllMocks()` and `unmockkAll()` exist: they prevent these leaks. They're part of the regular MockK API.

Because this PR is old and I've been working with MockK in many other projects, in the meantime I've created a library for precisely this. 😄 I will update this PR so that it no longer conflicts with the base branch, and I'll introduce and use my library. I'll request your review again when done.

See also [the MockK readme](https://github.com/mockk/mockk#clearing-vs-unmocking) and [my messages about this on the MockK channel in the Kotlin Slack](https://kotlinlang.slack.com/archives/C9EJFT6DB/p1583918385041900).

Reply:
Ok, is it specific to uniflow then? 🤔

Reply:
Not at all specific to uniflow. MockK just requires some boilerplate (depending on how you use it). That boilerplate is reduced a lot by my lib.